simple_history/admin.py

```diff
@@ -103,6 +111,29 @@ def history_view_title(self, request, obj):
         else:
             return _("Change history: %s") % force_str(obj)
 
+    def set_history_delta_changes(self, action_list: Sequence[HistoricalChanges]):
+        # Add a `history_delta_changes` attribute to all history records
+        # except the first (oldest) one
+        for i in range(len(action_list) - 1):
+            delta = action_list[i].diff_against(action_list[i + 1])
+            action_list[i].history_delta_changes = [
+                self.format_history_delta_change(change) for change in delta.changes
+            ]
```

Comment:
```suggestion
        previous = None
        current = None
        for current in action_list:
            if previous is None:
                previous = current
                continue
            delta = previous.diff_against(current)
            previous.history_delta_changes = [
                self.format_history_delta_change(change)
                for change in delta.changes
            ]
```

This allows us to iterate `action_list` just once rather than technically twice with several index operations.

Reply:
Thanks, that should definitely make it more efficient! I personally think the original version is slightly easier to understand, but your version could make a noticeable difference for large history tables, so let's go with that 🙂

Reply:
Is there a particular section of it that could benefit from a comment or two?

Reply:
Not that I can think of 🤔 I guess it's just that I have to pay slightly more attention to what `previous` and `current` are and when they're set and used, but it's such a small algorithm that I think the existing comment (at the top of the method) does the job 🙂